gig/views.py

```diff
@@ -114,11 +116,33 @@ def form_valid(self, form):
 
         return result
 
+class DuplicateView(CreateView):
+  
+    def get_form_kwargs(self, *args, **kwargs):
+        kwargs = super().get_form_kwargs(*args, **kwargs)
+        gig_orig = Gig.objects.get(id=self.kwargs['pk'])
+        # we didn't originally get a band in the request, we got a gig pk - so get the band from that
+        # and stash it among the args from the request
+        self.kwargs['bk'] = gig_orig.band.id
+
+        # populate the initial data from the original gig
+        kwargs['initial'] = forms.models.model_to_dict(gig_orig, 
+                                                        exclude=['calldate', 'setdate', 'enddate'])
+        # ...but replace the title with a 'copy of'
+        kwargs['initial']['title'] = f'Copy of {kwargs["initial"]["title"]}'
+        return kwargs
+
+    def get_band_from_kwargs(self, **kwargs):
+        # didn't have the band from the request args, so pull it from the gig
+        try:
+            return Gig.objects.get(id=self.kwargs['pk']).band
+        except Gig.DoesNotExist:
+            raise Http404('No gig found matching the query')
```

Comment:
Perhaps a place to use `get_object_or_404`.

Reply:
"perhaps" - you're too nice